### PR TITLE
Fixed #27420 -- Quoted Oracle DB test user password

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

**Comment by charettes** [comment]
What if the test password contains a `"` character? This should either be escaped at the driver level (`cx_Oracle`) or relying on something like `self.connection.schema_editor.quote_value`.

**Comment by felixxm** [comment]
User password in Oracle cannot contain quotation marks (see [Oracle documentation](http://docs.oracle.com/database/121/SQLRF/statements_8003.htm#SQLRF54736)).

**Comment by timgraham** [comment]
buildbot, test on oracle.

**Comment by timgraham** [comment]
Some Oracle builds on Jenkins are giving "ORA-28001: the password has expired" since da7910d4834726eca596af0a830762fa5fb2dfd9. I guess that's probably unrelated to this fix. \cc @intgr

**Comment by felixxm** [comment]
I unlocked Oracle DB user and made his password non-expiring.

**Comment by intgr** [comment]
Ouch, sorry about this bug. :(

About password expiry: do you think there is any connection between the test password policy changes in Django and the expiration? I can't see a reason why changing the password would make it expire sooner -- if anything, it should delay expiration...

What was the `PASSWORD_LIFE_TIME` set to? You can use this query...
```sql
select resource_name,limit from dba_profiles where profile=(
    select profile from dba_users where username='TEST_USERNAME'
);
```

**Comment by felixxm** [comment]
It's just a coincidence that password expired on the 1st of November.

**Comment by felixxm** [comment]
@timgraham Please restart test on oracle.

**Comment by timgraham** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
Patch improvement due to [comment](https://code.djangoproject.com/ticket/27420#comment:6).

File: django/db/backends/oracle/creation.py [hunk 1]
```diff
@@ -77,9 +77,6 @@ def _create_test_db(self, verbosity=1, autoclobber=False, keepdb=False):
             try:
                 self._create_test_user(cursor, parameters, verbosity, keepdb)
             except Exception as e:
-                # If we want to keep the db, then we want to also keep the user.
-                if keepdb:
-                    return
```
Comment (intgr) [inline django/db/backends/oracle/creation.py]: +1 to this, thanks!

Comment (timgraham) [inline django/db/backends/oracle/creation.py]: Should this be treated as part of the same ticket? I'm not sure if there's a risk of a regression if it's backported to the stable branches.

Comment (felixxm) [inline django/db/backends/oracle/creation.py]: IMO the only situation in which `_create_test_user` can raise exception that we want to omit (when `--keepdb` flag is on) is already checked in this [line](https://github.com/django/django/blob/master/django/db/backends/oracle/creation.py#L225) (`ORA-01920: user name 'string' conflicts with another user or role name`).

Comment (timgraham) [inline django/db/backends/oracle/creation.py]: Okay, but should this have been part of the original security fix? While it hid a problem, it doesn't seem related.

Comment (felixxm) [inline django/db/backends/oracle/creation.py]: Ok, I prepared separate [PR](https://github.com/django/django/pull/7531) for this.

**Comment by timgraham** [comment]
merged in c4b04e1598c4325454c808183dce17b284ed9e28, thanks!
